### PR TITLE
feat: allow overriding devnet upgrade heights with env vars

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Lotus allows overriding upgrade heights with `LOTUS_XX_HEIGHT` environment variables on 2k devnet, this PR tries to implement the same feature for forest devnet, using `FOREST_XX_HEIGHT` environment variables.  The feature is covered by `devnet checks` CI job.

Note: ~It's based on https://github.com/ChainSafe/forest/pull/3895 and should let https://github.com/ChainSafe/forest/pull/3895 go first~

Changes introduced in this pull request:

- allow customing devnet upgrade heights with `FOREST_XX_HEIGHT` environment variables

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/networks/devnet/mod.rs [hunk 1]
```diff
@@ -6,132 +6,134 @@ use once_cell::sync::Lazy;
...
         HeightInfo {
             height: Height::Thunder,
-            epoch: -1,
+            epoch: get_upgrade_height("FOREST_THUNDER_HEIGHT", -1),
             bundle: None,
         },
         HeightInfo {
             height: Height::Watermelon,
-            epoch: 10,
+            epoch: get_upgrade_height("FOREST_WATERMELON_HEIGHT", 200),
```
Comment (LesnyRumcajs) [inline src/networks/devnet/mod.rs:136]: Perhaps it'd be more idiomatic to have it as `Result` so the usage would be `get_upgrade_height("FOREST_WATERMELON_HEIGHT").unwrap_or(200)`. Not a big deal, though.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/networks/devnet/mod.rs (hunk 1 above)
Comment (hanabi1224) [reply src/networks/devnet/mod.rs:136]: I feel `get_upgrade_height("FOREST_WATERMELON_HEIGHT").unwrap_or(200)` suggests `FOREST_WATERMELON_HEIGHT` is defined in normal cases, so I'd prefer keeping it this way, just like lotus.

```go
getUpgradeHeight := func(ev string, def abi.ChainEpoch) abi.ChainEpoch {
		hs, found := os.LookupEnv(ev)
		if found {
			h, err := strconv.Atoi(hs)
			if err != nil {
				log.Panicf("failed to parse %s env var", ev)
			}

			return abi.ChainEpoch(h)
		}

		return def
	}
```

Comment (LesnyRumcajs) [reply src/networks/devnet/mod.rs:136]: Lotus has clunky Go error handling and C-like notion of optional values. How about an `Option<ChainEpoch>`?

Comment (hanabi1224) [reply src/networks/devnet/mod.rs:136]: As for the function name `get_upgrade_height` I'd prefer `get_upgrade_height(&str, i64) -> i64`, or I can rename it to `get_upgrade_height_from_env_var(&str) -> Option<i64>`, what do you think?

Comment (LesnyRumcajs) [reply src/networks/devnet/mod.rs:136]: IMO the 2nd one is more expressive and leaves less room for guessing. I'd omit the `_var` in the name.

Comment (hanabi1224) [reply src/networks/devnet/mod.rs]: @LesnyRumcajs fixed

**Review by sudo-shashank — APPROVED** [review]

**Pull request merged**